vespa-application-maven-plugin/src/main/java/com/yahoo/container/plugin/mojo/ApplicationMojo.java

```diff
@@ -56,15 +57,30 @@ public void execute() throws MojoExecutionException {
 
     /** Writes meta data about this package if the destination directory exists. */
     private void addBuildMetaData(File applicationDestination) throws MojoExecutionException {
-        if ( ! applicationDestination.exists())
-            return;
+        if ( ! applicationDestination.exists()) return;
 
-        if (vespaversion == null) // Get the build version of the parent project unless specifically set.
-            vespaversion = project.getProperties().getProperty("vespaversion");
+        if (vespaversion == null)
+            vespaversion = project.getPlugin("com.yahoo.vespa:vespa-application-maven-plugin").getVersion();
 
-        String metaData = String.format("{\"compileVersion\": \"%s\",\n \"buildTime\": %d}",
-                                        vespaversion,
-                                        System.currentTimeMillis());
+        Version compileVersion = Version.from(vespaversion);
+
+        MavenProject current = project;
+        while (current.getParent() != null && current.getParent().getParentArtifact() != null)
+            current = current.getParent();
+
+        boolean hasVespaParent = false;
+        Artifact parentArtifact = current.getParentArtifact();
+        if (parentArtifact != null && parentArtifact.getGroupId().startsWith("com.yahoo.vespa")) {
```

Comment:
Should we add a future proof `startsWith(ai.vespa.)` as well?

Suggest to add a trailing dot to com.yahoo.vespa anyway.

Reply:
Good points, fixed.